compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -197,66 +235,70 @@ LogicalResult verifyDoubleTilingExpertPassPipelineConfig(
       }
     }
 
-    SmallVector<int64_t> secondLevelTileSizes = loweringConfig.getTileSizeVals(
-        static_cast<unsigned>(TilingLevel::ParallelTiles));
+    SmallVector<int64_t> secondLevelTileSizes =
+        tilingConfig.getParallelVectorSizes();
     for (auto [index, tileSize] : llvm::enumerate(secondLevelTileSizes)) {
       if (tileSize != 0 && !pLoopsSet.contains(index)) {
         return op->emitOpError(
-                   "expected only parallel dims to be set in the "
-                   "second tiling sizes, got ")
+                   "expected only parallel dims to be set in the second tiling "
+                   "level, got ")
                << index << "-th tile size set";
       }
     }
 
-    SmallVector<int64_t> thirdLevelTileSizes = loweringConfig.getTileSizeVals(
-        static_cast<unsigned>(TilingLevel::ReductionTiles));
+    SmallVector<int64_t> thirdLevelTileSizes =
+        tilingConfig.getReductionVectorSizes();
     for (auto [index, tileSize] : llvm::enumerate(thirdLevelTileSizes)) {
       if (tileSize != 0 && pLoopsSet.contains(index)) {
         return op->emitOpError(
-                   "expected only reduction dims to be set in the third "
-                   "tiling sizes, got ")
+                   "expected only reduction dims to be set in the third tiling "
+                   "level, got ")
                << index << "-th tile size set";
       }
     }
   }
 
-  // Verify interchange
-  if (!loweringConfig.getTileInterchange().empty()) {
-    for (auto level : llvm::seq<unsigned>(
-             0, static_cast<unsigned>(
-                    loweringConfig.getTileInterchange().size()))) {
-      auto tileSizes = loweringConfig.getTileSizeVals(level);
-      auto interchange = loweringConfig.getTileInterchangeVals(level);
-      if (!isValidInterchange(interchange, tileSizes.size())) {
-        return op->emitOpError("expected [0, ")
-               << tileSizes.size()
-               << ") to be set exactly once in interchange #" << level;
-      }
-    }
-  }
-
-  // Verify that native vector size is empty.
-  SmallVector<int64_t> nativeVectorSize =
-      loweringConfig.getNativeVectorSizeVals();
-  if (!nativeVectorSize.empty()) {
-    return op->emitOpError("native_vector_size must be empty");
-  }
+  // TODO: What is this?
+  //// Verify interchange
+  // if (!loweringConfig.getTileInterchange().empty()) {
+  //   for (auto level : llvm::seq<unsigned>(
+  //            0, static_cast<unsigned>(
+  //                   loweringConfig.getTileInterchange().size()))) {
+  //     auto tileSizes = loweringConfig.getTileSizeVals(level);
+  //     auto interchange = loweringConfig.getTileInterchangeVals(level);
+  //     if (!isValidInterchange(interchange, tileSizes.size())) {
+  //       return op->emitOpError("expected [0, ")
+  //              << tileSizes.size()
+  //              << ") to be set exactly once in interchange #" << level;
```

Comment:
Any feedback about these fields in lowering config? I've never seen a "native vector size" or "tile interchange" in there. Should I add them to the TilingConfig API?

Reply:
nativeVectorSize should be deprecated in these pipeline. They were used for vector level tiling.

tile_interchange is an experimental path, we don't use it by default. We provided the option for tuning, but we did not get a performance boost from it. Example usage: https://github.com/openxla/iree/blob/35e97ecb776dab96d6d4f27daf2cabe63f183351/tests/e2e/regression/lowering_config.mlir#L9-L12